### PR TITLE
Add support for Win 7, 8 10 (<17663) to NET5

### PR DESCRIPTION
**Pull request by bgavrilMS**
Please see https://docs.microsoft.com/en-us/dotnet/standard/analyzers/platform-compat-analyzer and 
    https://github.com/dotnet/designs/blob/main/accepted/2020/platform-checks/platform-checks.md

### CODE REVIEW

File: build/SolutionWideAnalyzerConfig.ruleset [hunk 1]
```diff
@@ -79,7 +79,7 @@
   </Rules>
   <Rules AnalyzerId="Microsoft.NetCore.Analyzers" RuleNamespace="Microsoft.NetCore.Analyzers">
     <Rule Id="CA1304" Action="Error" />
-    <Rule Id="CA1305" Action="Warning" />
+    <Rule Id="CA1305" Action="None" />
```
Comment (bgavrilMS) [inline build/SolutionWideAnalyzerConfig.ruleset:82]: unrelated..

File: src/client/Microsoft.Identity.Client/Microsoft.Identity.Client.csproj [hunk 2]
```diff
@@ -71,9 +71,13 @@
...
-    <!-- Remove after Dec Release of .NET SDK -->
-    <WindowsSDKVersion>10.0.17763.0</WindowsSDKVersion>
     <UseWindowsForms>true</UseWindowsForms>
+    <!-- 
+    MSAL will run on Windows 7 and 8 but requires to be built against Win10 to use WinRT APIs for WAM 
+    See https://docs.microsoft.com/en-us/dotnet/standard/analyzers/platform-compat-analyzer and 
+    https://github.com/dotnet/designs/blob/main/accepted/2020/platform-checks/platform-checks.md for details
+    -->
+    <SupportedOSPlatformVersion>7</SupportedOSPlatformVersion>
```
Comment (bgavrilMS) [inline src/client/Microsoft.Identity.Client/Microsoft.Identity.Client.csproj:80]: Functionally, this is all that is needed.

File: src/client/Microsoft.Identity.Client/Platforms/Features/DesktopOS/StaTaskScheduler.cs [hunk 3]
```diff
@@ -13,6 +13,9 @@ namespace Microsoft.Identity.Client.Platforms.Features.DesktopOs
     // This IDisposable class doe not need to implement Dispose method in standard way, because it is sealed.
     // If it ever needs to become inheritable, it should follow the standard pattern as desribed in http://msdn.microsoft.com/en-us/library/fs2xkftw(v=vs.110).aspx.
     /// <summary>Provides a scheduler that uses STA threads.</summary>
+#if NET5_WIN
+    [System.Runtime.Versioning.SupportedOSPlatform("windows")]
```
Comment (bgavrilMS) [inline src/client/Microsoft.Identity.Client/Platforms/Features/DesktopOS/StaTaskScheduler.cs:17]: Don't know why the analyzer complains as all these types are internal... Logged a bug on .net team ... https://github.com/dotnet/roslyn-analyzers/issues/4933 but for now, annotating the warnings.

**Comment by bgavrilMS** [comment]
CC @mjcheetham

File: src/client/Microsoft.Identity.Client/AppConfig/PublicClientApplicationBuilder.cs [hunk 4]
```diff
@@ -135,19 +135,33 @@ public PublicClientApplicationBuilder WithIosKeychainSecurityGroup(string keycha
...
                 throw new PlatformNotSupportedException(
-                    "The Windows broker is not directly available on MSAL for .NET Framework or .NET Core 3.x" +
+                    "The Windows broker is not directly available on MSAL for .NET Framework" +
                     " To use it, please install the nuget package named Microsoft.Identity.Client.Desktop " +
-                    "and call the extension method .WithWindowsBroker(true)");
+                    "and call the extension method .WithDesktopFeatures() first.");
+            }
+#endif
+
+#if NET_CORE
```
Comment (bgavrilMS) [inline src/client/Microsoft.Identity.Client/AppConfig/PublicClientApplicationBuilder.cs:155]: @mjcheetham - does this look right to you? When you target .net5 on Mac and Linux, it will NO-OP for now (until we have broker on Mac / Linux). On Windows, it will throw this exception.

File: src/client/Microsoft.Identity.Client/Platforms/Features/DesktopOS/StaTaskScheduler.cs (hunk 3 above)
Comment (pmaytak) [inline src/client/Microsoft.Identity.Client/Platforms/Features/DesktopOS/StaTaskScheduler.cs:17]: Why do we need to specify this? Is the default support 17663+ because of our target?

My understanding if I run MSAL on Win 7, then all these classes are available, unless they are marked with `SupportedOSPlatform("windows10.0.17763.0")`, which makes them available on 17763+.

Or is this to specify that this is available only on windows and not linux and such?

**Review by pmaytak — COMMENTED** [review]
LGTM (although I admit it takes a bit time to get used to these different version combinations).

Is my understanding correct?
- We target net5 17663 to use WAM APIs.
- `<SupportedOSPlatformVersion>7</SupportedOSPlatformVersion>` allows MSAL to run on Windows 7+.
- But now we have to use `SupportedOSPlatform` to hide the 17663+ APIs from older runtimes.

File: src/client/Microsoft.Identity.Client/Platforms/Features/DesktopOS/StaTaskScheduler.cs (hunk 3 above)
Comment (bgavrilMS) [reply src/client/Microsoft.Identity.Client/Platforms/Features/DesktopOS/StaTaskScheduler.cs:17]: It looks like 

`thread.SetApartmentState(ApartmentState.STA);`

is available only on Windows. 

I don't know why the analyzer is complaining about this, since net5-windows10.xyz is only available on Windows :) 
I'll log another bug on them I think.
